library/src/main/java/com/smartnsoft/droid4me/download/gif/Gif.java

```diff
@@ -0,0 +1,202 @@
+package com.smartnsoft.droid4me.download.gif;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import android.animation.ValueAnimator;
+import android.animation.ValueAnimator.AnimatorUpdateListener;
+import android.graphics.Bitmap;
+import android.os.Build.VERSION_CODES;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.view.View;
+import android.view.View.OnAttachStateChangeListener;
+import android.view.animation.LinearInterpolator;
+import android.widget.ImageView;
+
+import com.smartnsoft.droid4me.log.Logger;
+import com.smartnsoft.droid4me.log.LoggerFactory;
+
+
+/**
+ * @author Antoine Gerard
+ * @since 2017.10.18
+ */
+
+/**
+ * This class represents a Gif File and its display in an ImageView
+ */
+@RequiresApi(api = VERSION_CODES.HONEYCOMB_MR1)
+public final class Gif
+    implements OnAttachStateChangeListener
+{
+
+  public static final Logger log = LoggerFactory.getInstance("Gif");
+
+  private final List<Bitmap> bitmaps;
+
+  private int duration;
+
+  private ImageView imageView;
+
+  /**
+   * We are using a value animator for animating
+   * the gif in the ImageView
+   */
+  private ValueAnimator valueAnimator;
+
+
+  public Gif(InputStream inputStream, String url)
+  {
+    this.bitmaps = new ArrayList<>();
+    decodeGif(inputStream, url);
+  }
+
+  @Override
+  public void onViewAttachedToWindow(View view)
+  {
+
+  }
+
+  /**
+   * We end the gif animation once the ImageView that plays the
+   * gif is detached from the current window.
+   * This will prevent multiple display of the same gif in a recycler view
+   * It will also free the memory preventing memory leak and OOM
+   *
+   * @param view
+   */
+  @Override
+  public void onViewDetachedFromWindow(View view)
+  {
+    end();
```

Comment:
Ne faudrait t-il pas rajouter     this.imageView.removeOnAttachStateChangeListener(this);
pour éviter un leak ?